lib/core/utils/pref_utils.dart

```diff
@@ -0,0 +1,23 @@
+//ignore: unused_import
+import 'dart:convert';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class PrefUtils {
+  static SharedPreferences? _sharedPreferences;
+
+  PrefUtils() {
+    SharedPreferences.getInstance().then((value) {
+      _sharedPreferences = value;
+    });
+  }
+
+  Future<void> init() async {
+    _sharedPreferences ??= await SharedPreferences.getInstance();
+    print('SharedPreference Initialized');
+  }
+
+  ///will clear all the data stored in preference
+  void clearPreferencesData() async {
+    _sharedPreferences!.clear();
```

Comment:
force unwrapping is not a good practice if it could be an optional then use null safety the above could be replaced with `_sharedPreferences?.clear();`

```suggestion
    _sharedPreferences?.clear();
```